### ISSUE TITLE
REGRESSION: network_cli with ios hanging

### ISSUE THREAD

**Issue by coreywan** [ansible/ansible#43031]
##### SUMMARY

network_cli is hanging when trying to connect to our ios devices.  We did a packet capture, and the ansible control host is not even trying to make a connection to the network device.

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
(ansible2.6) C02VV262HV2R:ansible-network_remote_site wanlessc$ ansible --version
ansible 2.6.1
  config file = /Users/wanlessc/innovate/ansible-network_remote_site/ansible.cfg
  configured module search path = [u'/Users/wanlessc/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wanlessc/miniconda2/envs/ansible2.6/lib/python2.7/site-packages/ansible
  executable location = /Users/wanlessc/miniconda2/envs/ansible2.6/bin/ansible
  python version = 2.7.15 |Anaconda, Inc.| (default, May  1 2018, 18:37:05) [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
```

##### CONFIGURATION

(ansible2.6) C02VV262HV2R:ansible-network_remote_site wanlessc$ ansible-config dump --only-changed
DEFAULT_VAULT_PASSWORD_FILE(**path to config** ) = **path to file**

##### OS / ENVIRONMENT

Network Device Version: 03.06.08E
MAC is the control host (10.12.6)

##### STEPS TO REPRODUCE

Playbook:
```yaml
- hosts: rs_access_switch
  gather_facts: false
  connection: network_cli
  tasks:
    - ios_command:
        commands: show interface
```
group_vars:
```yaml
ansible_user: my_super_secret_admin_account_pass_is_pass123
ansible_password: super_secret
ansible_network_os: ios
```

##### EXPECTED RESULTS

For it to connect to the switch
##### ACTUAL RESULTS

When in verbose mode, it just sits at this "using connection plugin network_cli" and never moves forward. If I jump back to ansible 2.5.6 it works fine.

```
(ansible2.6) C02VV262HV2R:ansible-network_remote_site wanlessc$ ansible-playbook -i inventories/tst/hosts -C -vvvvv bug.yaml
ansible-playbook 2.6.1
  config file = /Users/wanlessc/innovate/ansible-network_remote_site/ansible.cfg
  configured module search path = [u'/Users/wanlessc/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wanlessc/miniconda2/envs/ansible2.6/lib/python2.7/site-packages/ansible
  executable location = /Users/wanlessc/miniconda2/envs/ansible2.6/bin/ansible-playbook
  python version = 2.7.15 |Anaconda, Inc.| (default, May  1 2018, 18:37:05) [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
Using /Users/wanlessc/innovate/ansible-network_remote_site/ansible.cfg as config file
Reading vault password file: /Users/wanlessc/network.pass
setting up inventory plugins
Parsed /Users/wanlessc/innovate/ansible-network_remote_site/inventories/tst/hosts inventory source with ini plugin
Loading callback plugin default of type stdout, v2.0 from /Users/wanlessc/miniconda2/envs/ansible2.6/lib/python2.7/site-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: bug.yaml ******************************************************************************************************************************************
1 plays in bug.yaml

PLAY [rs_access_switch] *************************************************************************************************************************************
META: ran handlers

TASK [ios_command] ******************************************************************************************************************************************
task path: /Users/wanlessc/innovate/ansible-network_remote_site/bug.yaml:5
Found a vault_id (default) in the vaulttext
We have a secret associated with vault id (default), will try to use to decrypt None
Trying to use vault secret=(FileVaultSecret(filename='/Users/wanlessc/network.pass')) id=default to decrypt None
Trying secret FileVaultSecret(filename='/Users/wanlessc/network.pass') for vault_id=default
Decrypt successful with secret=FileVaultSecret(filename='/Users/wanlessc/network.pass') and vault_id=default
Found a vault_id (default) in the vaulttext
We have a secret associated with vault id (default), will try to use to decrypt None
Trying to use vault secret=(FileVaultSecret(filename='/Users/wanlessc/network.pass')) id=default to decrypt None
Trying secret FileVaultSecret(filename='/Users/wanlessc/network.pass') for vault_id=default
Decrypt successful with secret=FileVaultSecret(filename='/Users/wanlessc/network.pass') and vault_id=default
<sw01-remotesite.wwt.com> attempting to start connection
<sw01-remotesite.wwt.com> using connection plugin network_cli
```

**Comment by coreywan** [comment]
This appears to have been fixed in the 2.6.3 release

**Comment by iwasinnamuknow** [comment]
I'm seeing this exact problem with 2.7.6. Some details:

```
ansible-playbook 2.7.6
python version = 2.7.13
```

```
ansible_connection: network_cli
ansible_network_os: ios
ansible_user: me
ansible_ssh_pass: mypass
ansible_become: yes
ansible_become_method: enable
ansible_become_pass: enpass
```

```
 hosts: switches
  gather_facts: no

  tasks:
    - name: Show interface test
      ios_command:
        command: show interface
```

It hangs here:

```
TASK [Show interface test] ******************************************
<10.1.4.1> attempting to start connection
<10.1.4.1> using connection plugin network_cli
```

Should I start a new issue for this?
